packages/e2e-tests/test/test-shell.ts

```diff
@@ -44,8 +46,11 @@ export interface TestShellOptions {
  * Test shell helper class.
  */
 export class TestShell {
-  private static _openShells: TestShell[] = [];
+  private static _openShells: Set<TestShell> = new Set();
 
+  /**
+   * @deprecated Use the {@link Mocha.Context.startTestShell} hook instead
+   */
```

Comment:
I don't think we need to deprecate this, there shouldn't be anything _wrong_ with using it apart from accepting the responsibility to manually clean the instance up afterwards.

I'd maybe even consider making this "layering" a bit more explicit, and split out the mocha-specific parts into its own file, so that it's clear that `TestShell` is now a self-contained class that contains the parts that are independent from mocha usage?

Reply:
Actually, thinking about it, do we maybe even want to implement `Symbol.asyncDispose` here and use `await using shell = this.startTestShell(...)`?

Reply:
> there shouldn't be anything *wrong* with using it apart from accepting the responsibility to manually clean the instance up afterwards

I see your point and agree there could be future scenarios where one would want to start the shell directly.

However, I was successful in updating all direct calls to `TestServer.start` to use `startTestShell` instead (making it a good candidate for the preferred way of starting test shells) and this is the most effective way I know to nudge myself into the `startTestShell`-way and clearly signal to future me, that:

1. Although you can still call `TestShell.start` (since it isn't private), it is not the preferred way.
2. You should be careful if you do call `TestShell.start` yourself, as you take on responsibilities.

Do you have any suggestions for alternatives to signal this intent? I was considering a `console.warn`, but that easily gets lost in the output and doesn't give signal when the code is being written.

> split out the mocha-specific parts into its own file

Makes sense 👍 I've pushed a commit splitting this and updated the PR description.

Reply:
> Do you have any suggestions for alternatives to signal this intent?

Honestly, I think just a regular doc comment does the trick here perfectly well 🙂

Reply:
Let's try 👍 I've pushed a commit with good ol' doc comment 🙂